### PR TITLE
Gh6566 2 customer retention: repair logic, refactoring, test

### PR DESCRIPTION
**Pull request by metas-rc**
https://github.com/metasfresh/metasfresh/issues/6566

### CODE REVIEW

File: backend/de.metas.contracts/src/main/java/de/metas/contracts/impl/CustomerRetentionRepository.java [hunk 1]
```diff
@@ -64,13 +68,18 @@
 	private final IInvoiceDAO invoiceDAO = Services.get(IInvoiceDAO.class);
 	private final IContractsDAO contractsDAO = Services.get(IContractsDAO.class);
 	private final ContractInvoiceService contractInvoiceService;
-	
+
+	public I_C_Customer_Retention getById(final CustomerRetentionId customerRetentionId)
```
Comment (metas-ts) [inline backend/de.metas.contracts/src/main/java/de/metas/contracts/impl/CustomerRetentionRepository.java:72]: pls add `@NonNull` at least to public methods like this one

File: backend/de.metas.contracts/src/main/java/de/metas/contracts/impl/CustomerRetentionRepository.java [hunk 2]
```diff
@@ -80,134 +89,111 @@ public I_C_Customer_Retention createNewCustomerRetention(@NonNull final BPartner
...
+	@VisibleForTesting
+	void setNewCustomer(@NonNull final CustomerRetentionId customerRetentionId)
 	{
-		I_C_Customer_Retention customerRetention = retrieveCustomerRetention(bpartnerId);
-
-		if (customerRetention == null)
-		{
-			customerRetention = createNewCustomerRetention(bpartnerId);
-		}
+		final I_C_Customer_Retention customerRetention = getById(customerRetentionId);
```
Comment (metas-ts) [inline backend/de.metas.contracts/src/main/java/de/metas/contracts/impl/CustomerRetentionRepository.java:99]: please add a comment or better an Check.assumeNotNull here to make it explicit the you made sure elsewhere this won't be null

PS: if you need to rely on something outside this class, such as the model interceptors, i suggest to invoke that getCreate method instead..

File: backend/de.metas.contracts/src/main/java/de/metas/contracts/impl/CustomerRetentionRepository.java (hunk 1 above)
Comment (metas-rc) [reply backend/de.metas.contracts/src/main/java/de/metas/contracts/impl/CustomerRetentionRepository.java:72]: Done.

File: backend/de.metas.contracts/src/main/java/de/metas/contracts/impl/CustomerRetentionRepository.java (hunk 2 above)
Comment (metas-rc) [reply backend/de.metas.contracts/src/main/java/de/metas/contracts/impl/CustomerRetentionRepository.java:99]: Check.assumeNotNull done.

**Pull request merged**